src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
-        return null;
+        Node<T> node = findNodeByIndex(index);
+        T value = node.item;
+        unlink(node);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = getNode(object);
+        if (node == null) {
+            return false;
+        }
+        unlink(node);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addCheck(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private void indexCheck(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private Node<T> getNode(T value) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if ((node.item == value) || (node.item != null && node.item.equals(value))) {
+                return node;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        indexCheck(index);
+        if ((size / 2) > index) {
+            Node<T> node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+            return node;
+        }
+        Node<T> node = tail;
+        for (int i = 0; i < size - index - 1; i++) {
+            node = node.prev;
+        }
+        return node;
+    }
+
+    private void linkBefore(T element, Node<T> next) {
+        Node<T> prev = next.prev;
+        Node<T> newNode = new Node<>(prev, element, next);
+        next.prev = newNode;
+        if (prev == null) {
+            head = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private void setFirstValue(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        head = tail = newNode;
+        size++;
+    }
+
+    private void linkLast(T element) {
+        Node<T> newNode = new Node<>(tail, element, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+    }
+
+    private void unlink(Node<T> node) {
+        size--;
+        if (node == head && node == tail) {
+            head = null;
+            tail = null;
+            return;
+        } else if (node == head) {
+            head = node.next;
+            node.next.prev = null;
+            return;
+        } else if (node == tail) {
+            tail = node.prev;
+            node.prev.next = null;
+            return;
+        }
+        node.next.prev = node.prev;
+        node.prev.next = node.next;
```

Comment:
I'd recommend you to put this inside `else` block, `size--;` put after `else` and delete all `return` from this method. That will optimize your code.